### PR TITLE
[Docs] Remove Conan from documentation

### PR DESCRIPTION
**Pull request by hjabird**
# Description

* Support for Conan has been dropped in oneMKL. See https://github.com/oneapi-src/oneMKL/discussions/267.
* This PR:
  * Removes Conan from the documentation.
  * Removes Conan scripts.

This is one of the suggested changes in https://github.com/oneapi-src/oneMKL/issues/459.

# Checklist

## All Submissions

- [ N/A] Do all unit tests pass locally? Attach a log.
  - I have confirmed that it is still possible to build the project and that the CMake has not been broken. I have not tested the C++ code since no changes have been made.
- [ N/A] Have you formatted the code using clang-format?

### CODE REVIEW

**Comment by hjabird** [comment]
Pinging @mkrainiuk @mmeterel @vmalia since I think you're best placed to know what we want here.

File: README.md [hunk 1]
```diff
@@ -476,80 +468,35 @@ Microsoft Windows* Server | 2016, 2019 | *Not supported*
...
             <td> AMD GPU </td>
             <td> oneAPI DPC++ Compiler <br> or <br> AdaptiveCpp with ROCm backend and dependencies </td>
-            <td> No </td>
         </tr>
     </tbody>
 </table>
 
-*If [Building with Conan](https://oneapi-src.github.io/oneMKL/building_the_project.html#building-with-conan), above packages marked as "No" must be installed manually.*
-
 *If [Building with CMake](https://oneapi-src.github.io/oneMKL/building_the_project.html#building-with-cmake), above packages must be installed manually.*
```
Comment (Rbiessy) [inline README.md]: That sentence looks weird now. The table under "General" should mention that downloading cmake is mandatory now and potentially link to build_the_project. I think that sentence can be removed.

Comment (hjabird) [reply README.md]: I added the CMake requirement to the table in [6f0a204](https://github.com/oneapi-src/oneMKL/pull/460/commits/6f0a204e20de281134417690da23212d4cca4384) and deleted the mentioned sentence in [96014f4](https://github.com/oneapi-src/oneMKL/pull/460/commits/96014f4e502a39db57ca08ea49b9df74264da673).

**Review by Rbiessy — APPROVED** [review]

**Review by mkrainiuk — APPROVED** [review]
Thanks for the changes! Looks good to me. I believe we can also close the discussion about Conan after this change, what do you think @vmalia?

**Comment by hjabird** [comment]
I'll merge this tomorrow if there are no further comments (@vmalia?).

**Pull request merged**